client/vscode-lib/src/controller.ts

```diff
@@ -119,73 +103,88 @@ export function createController({
         providers,
     })
 
-    const errorTapObserver: Partial<TapObserver<any>> = {
-        next(): void {
-            errorWaiter.gotError(false)
-        },
-        error(): void {
-            // Show an error notification unless we've recently shown one (to avoid annoying
-            // the user).
-            const shouldNotify = errorWaiter.timeSinceLastError() > 1000 * 60 * 15 /* 15 min */
-            if (shouldNotify) {
-                showErrorNotification(outputChannel)
-            }
-
-            errorWaiter.gotError(true)
-        },
-    }
-    const errorCatcher = <T = any>(error: any): Observable<T[]> => {
-        outputChannel.appendLine(error)
-        return of([])
-    }
+    // Pause for 10 seconds if we get 5 errors in a row.
+    const errorDelay = 10 * 1000
+    const errorThreshold = 5
+    const errorWaiterMeta = createErrorWaiter(errorDelay, errorThreshold)
+    const errorWaiterMentions = createErrorWaiter(errorDelay, errorThreshold)
+    const errorWaiterItems = createErrorWaiter(errorDelay, errorThreshold)
+    const errorWaiterAnnotations = createErrorWaiter(errorDelay, errorThreshold)
+    disposables.push(errorWaiterMeta, errorWaiterMentions, errorWaiterItems, errorWaiterAnnotations)
+
+    const errorReporterMeta = createErrorReporter(
+        outputChannel,
+        errorWaiterMeta,
+        'OpenCtx provider failed initializing.'
```

Comment:
It might be nicer to make it one-per-provider, and make the message:

`Error initializing OpenCtx provider '${providerUrl}': ${errorMessage}`

With the actions:

"Show Logs", "Open Settings"

Bubbling up the error message would also really help not having to trawl through our noisy logs, esp for simple things like [misconfiguring a settings key](https://github.com/sourcegraph/openctx/blob/main/provider/jira/index.ts#L22).

And having "Open Settings", and/or a simple disable way to disable a misbehaving or misconfigured provider, reduces the risk them getting super annoyed with the notification and uninstalling Cody to make it go away.

"Open Settings" would open VS Code settings filtered to `@ext:sourcegraph.cody-ai openctx` similarly to
https://github.com/sourcegraph/cody/blob/0e31fd3a148c5715ebc249d49ee835d506e7caa4/vscode/src/services/StatusBar.ts#L204-L206

Similarly for all the other messages (suggestions below), actions buttons being the same